File: model/src/main/java/io/github/runtimemodels/chazm/entity/SpecificationGoalEntity.java
package io.github.runtimemodels.chazm.entity;

import io.github.runtimemodels.chazm.id.UniqueId;

import javax.inject.Inject;
import javax.validation.constraints.NotNull;

import com.google.inject.assistedinject.Assisted;

class SpecificationGoalEntity extends AbstractEntity<SpecificationGoal> implements SpecificationGoal {

	@Inject
	SpecificationGoalEntity(@NotNull @Assisted final UniqueId<SpecificationGoal> id) {
		super(id);
	}

	@Override
	public boolean equals(final Object object) {
		if (object instanceof SpecificationGoal) {
			final SpecificationGoal goal = (SpecificationGoal) object;
			return super.equals(goal);
		}
		return false;
	}

	@Override
	public int hashCode() {
		return super.hashCode();
	}

	@Override
	public String toString() {
		return super.toString();
	}

}

File: chazm-model/src/main/java/runtimemodels/chazm/model/entity/SpecificationGoalEntity.kt
package runtimemodels.chazm.model.entity

import com.google.inject.assistedinject.Assisted
import runtimemodels.chazm.api.entity.SpecificationGoal
import runtimemodels.chazm.api.id.UniqueId

import javax.inject.Inject
import javax.validation.constraints.NotNull

internal open class SpecificationGoalEntity @Inject constructor(@NotNull @Assisted id: UniqueId<SpecificationGoal>) : AbstractEntity<SpecificationGoal>(id), SpecificationGoal {

    override fun equals(other: Any?): Boolean {
        if (other is SpecificationGoal) {
            return super.equals(other)
        }
        return false
    }

    override fun hashCode(): Int = super.hashCode()

    override fun toString(): String = super.toString()

}
